src/components/PopupWithForm.js

```diff
@@ -17,15 +18,27 @@ export default class PopupWithForm extends Popup {
     return this._inputValues;
   }
 
+  setButtonText(submit, buttonText = "Saving...") {
+    if (submit) {
+      this._submitButton.textContent = buttonText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
+  }
+
   setEventListeners() {
     this._form.addEventListener("submit", () => {
       this._data = this._getInputValues();
-      console.log(this._data);
       this._handleFormSubmit(this._data);
     });
     super.setEventListeners();
   }
 
+  open() {
+    this._form.reset();
```

Comment:
Needs correction: You should reset the form when closing the popup, not when opening